core/src/test/scala/org/apache/spark/executor/ResourceDiscovererSuite.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.executor
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files => JavaFiles}
+import java.nio.file.attribute.PosixFilePermission._
+import java.util.EnumSet
+
+import com.google.common.io.Files
+
+import org.apache.spark._
+import org.apache.spark.internal.config._
+
+class ResourceDiscovererSuite extends SparkFunSuite
+  with LocalSparkContext {
+
+  test("Resource discoverer no gpus") {
+    val sparkconf = new SparkConf
+    val resources = ResourceDiscoverer.findResources(sparkconf, false)
+    assert(resources.get("gpu").isEmpty,
+      "Should have a gpus entry that is empty")
+  }
+
+  test("Resource discoverer multiple gpus") {
+    val sparkconf = new SparkConf
+
+    val file1 = File.createTempFile("test", "resourceDiscoverScript1")
+    try {
+      Files.write("echo 0,1", file1, StandardCharsets.UTF_8)
+      JavaFiles.setPosixFilePermissions(file1.toPath(),
```

Comment:
This might not work on Windows, but whatever. At worst we might have to skip it on Windows

Reply:
Is the preferred way to just use the Utils.isWindows and skip the test here?

Reply:
I think you can do that, with assume or whatever the other tests do. I'm pretty sure this won't work, for the stronger reason that the bash script you will create here won't run on Windows anyway (?)